### PR TITLE
python312Packages.pastedeploy: 3.0.1 -> 3.1

### PR DESCRIPTION
**Pull request by anthonyroussel**
## Description of changes

* https://github.com/Pylons/pastedeploy/compare/3.0.1...3.1

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by anthonyroussel** [comment]
Rebased against staging branch (500+ rebuilds)

File: pkgs/development/python-modules/paste/default.nix [hunk 1]
```diff
@@ -26,7 +26,9 @@ buildPythonPackage rec {
     patchShebangs tests/cgiapp_data/
   '';
 
-  propagatedBuildInputs = [
+  build-system = [ setuptools ];
+
+  dependencies = [
     setuptools
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/paste/default.nix:32]: Are you sure setuptools is required at runtime?

Comment (anthonyroussel) [reply pkgs/development/python-modules/paste/default.nix:32]: Thank you for the review!
Yes, `setuptools` is required for `pkg_resources`, see https://github.com/pasteorg/paste/blob/bc1753b00c5f342d28b1af73cd4f662b206c1f37/setup.py#L58

**Review by FliegendeWurst — APPROVED** [review]

File: pkgs/development/python-modules/pastedeploy/default.nix [hunk 2]
```diff
@@ -4,27 +4,30 @@
...
-    repo = pname;
+    repo = "pastedeploy";
     rev = "refs/tags/${version}";
     hash = "sha256-8MNeOcYPEYAfghZN/K/1v/tAAdgz/fCvuVnBoru+81Q=";
   };
 
   postPatch = ''
     substituteInPlace pytest.ini \
-      --replace " --cov" ""
+      --replace-fail " --cov" ""
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pastedeploy/default.nix:26]: We could also use pytest-cov-stub

File: pkgs/development/python-modules/pastedeploy/default.nix [hunk 3]
```diff
@@ -4,27 +4,30 @@
...
+  pyproject = true;
 
   disabled = pythonOlder "3.7";
 
   src = fetchFromGitHub {
     owner = "Pylons";
-    repo = pname;
+    repo = "pastedeploy";
     rev = "refs/tags/${version}";
     hash = "sha256-8MNeOcYPEYAfghZN/K/1v/tAAdgz/fCvuVnBoru+81Q=";
```
Comment (Shados) [inline pkgs/development/python-modules/pastedeploy/default.nix:21]: @anthonyroussel you didn't update the hash for the new version (which also means you built and tested against the old version's source, which happened to be in your local store as a fixed-output derivation).

Comment (anthonyroussel) [reply pkgs/development/python-modules/pastedeploy/default.nix:21]: Thank you for the review!

Oops sorry, indeed, looks like I forgot to update the hash for this package.

I created this PR to fix the hash: https://github.com/NixOS/nixpkgs/pull/368915

**Pull request merged**